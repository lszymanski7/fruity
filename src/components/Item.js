import PropTypes from 'prop-types'
import React, { useState } from 'react'

const Item = (props) => {
	const { id, name, img, checked, fruits, setFruits, setFilteredFruits } = props
	const [icon, setIcon] = useState(checked ? img.color : img.linear)
	const fruitsData = JSON.parse(JSON.stringify(fruits))

	const handleCheck = () => {
		const fruit = fruitsData.find((fruit) => fruit.id === id)
		fruit.checked = !fruit.checked
		setFruits(fruitsData)
		setIcon(fruit.checked ? img.color : img.linear)
		setFilteredFruits(fruitsData.filter((fruit) => fruit.checked === true))
	}

	return (
		<div className="item">
			<img className="item__image" id={id} src={icon} alt={name} onClick={handleCheck} />
			<p className="item__title">{name}</p>
		</div>
	)
}

Item.propTypes = {
	id: PropTypes.string.isRequired,
	name: PropTypes.string.isRequired,
	img: PropTypes.object.isRequired,
	checked: PropTypes.bool.isRequired,
	fruits: PropTypes.array.isRequired,
	setFruits: PropTypes.func.isRequired,
	setFilteredFruits: PropTypes.func.isRequired
}

export { Item as default }
