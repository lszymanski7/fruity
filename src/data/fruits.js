import { ICONS } from './constants'

// Fruits data
const fruits = [
    {
        id: 1,
        name: 'apple',
        img: ICONS.apple,
        checked: false
    },
    {
        id: 2,
        name: 'banana',
        img: ICONS.banana,
        checked: false
    },
    {
        id: 3,
        name: 'blackberry',
        img: ICONS.blackberry,
        checked: false
    },
    {
        id: 4,
        name: 'blueberries',
        img: ICONS.blueberries,
        checked: false
    },
    {
        id: 5,
        name: 'cherries',
        img: ICONS.cherries,
        checked: false
    },
    {
        id: 6,
        name: 'coconut',
        img: ICONS.coconut,
        checked: false
    },
    {
        id: 7,
        name: 'dragon fruit',
        img: ICONS.dragon_fruit,
        checked: false
    },
    {
        id: 8,
        name: 'fig',
        img: ICONS.fig,
        checked: false
    },
    {
        id: 9,
        name: 'grapes',
        img: ICONS.grapes,
        checked: false
    },
    {
        id: 10,
        name: 'kiwi',
        img: ICONS.kiwi,
        checked: false
    },
    {
        id: 11,
        name: 'lemon',
        img: ICONS.lemon,
        checked: false
    },
    {
        id: 12,
        name: 'lychee',
        img: ICONS.lychee,
        checked: false
    },
    {
        id: 13,
        name: 'mango',
        img: ICONS.mango,
        checked: false
    },
    {
        id: 14,
        name: 'melon',
        img: ICONS.melon,
        checked: false
    },
    {
        id: 15,
        name: 'orange',
        img: ICONS.orange,
        checked: false
    },
    {
        id: 16,
        name: 'papaya',
        img: ICONS.papaya,
        checked: false
    },
    {
        id: 17,
        name: 'peach',
        img: ICONS.peach,
        checked: false
    },
    {
        id: 18,
        name: 'pear',
        img: ICONS.pear,
        checked: false
    },
    {
        id: 19,
        name: 'pineapple',
        img: ICONS.pineapple,
        checked: false
    },
    {
        id: 20,
        name: 'plum',
        img: ICONS.plum,
        checked: false
    },
    {
        id: 21,
        name: 'pomegranate',
        img: ICONS.pomegranate,
        checked: false
    },
    {
        id: 22,
        name: 'star fruit',
        img: ICONS.star_fruit,
        checked: false
    },
    {
        id: 23,
        name: 'strawberry',
        img: ICONS.strawberry,
        checked: false
    },
    {
        id: 24,
        name: 'watermelon',
        img: ICONS.watermelon,
        checked: false
    }
]

export { fruits as default }
