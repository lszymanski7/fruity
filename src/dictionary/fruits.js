import { fruitsColor, fruitsLinear } from '../assets/icons/index'

const fruits = [{
    name: 'Apple',
    img: [
        fruitsColor['apple.png'].default,
        fruitsLinear['apple.png'].default
    ]
}, {
    name: 'Banana',
    img: [
        fruitsColor['banana.png'].default,
        fruitsLinear['banana.png'].default
    ]
}, {
    name: 'Blackberry',
    img: [
        fruitsColor['blackberry.png'].default,
        fruitsLinear['blackberry.png'].default
    ]
}, {
    name: 'Blueberries',
    img: [
        fruitsColor['blueberries.png'].default,
        fruitsLinear['blueberries.png'].default
    ]
}, {
    name: 'Cherries',
    img: [
        fruitsColor['cherries.png'].default,
        fruitsLinear['cherries.png'].default
    ]
}, {
    name: 'Coconut',
    img: [
        fruitsColor['coconut.png'].default,
        fruitsLinear['coconut.png'].default
    ]
}, {
    name: 'Dragon Fruit',
    img: [
        fruitsColor['dragon-fruit.png'].default,
        fruitsLinear['dragon-fruit.png'].default
    ]
}, {
    name: 'Fig',
    img: [
        fruitsColor['fig.png'].default,
        fruitsLinear['fig.png'].default
    ]
}, {
    name: 'Grapes',
    img: [
        fruitsColor['grapes.png'].default,
        fruitsLinear['grapes.png'].default
    ]
}, {
    name: 'Kiwi',
    img: [
        fruitsColor['kiwi.png'].default,
        fruitsLinear['kiwi.png'].default
    ]
}, {
    name: 'Lemon',
    img: [
        fruitsColor['lemon.png'].default,
        fruitsLinear['lemon.png'].default
    ]
}, {
    name: 'Lychee',
    img: [
        fruitsColor['lychee.png'].default,
        fruitsLinear['lychee.png'].default
    ]
}, {
    name: 'Mango',
    img: [
        fruitsColor['mango.png'].default,
        fruitsLinear['mango.png'].default
    ]
}, {
    name: 'Melon',
    img: [
        fruitsColor['melon.png'].default,
        fruitsLinear['melon.png'].default
    ]
}, {
    name: 'Orange',
    img: [
        fruitsColor['orange.png'].default,
        fruitsLinear['orange.png'].default
    ]
}, {
    name: 'Papaya',
    img: [
        fruitsColor['papaya.png'].default,
        fruitsLinear['papaya.png'].default
    ]
}, {
    name: 'Peach',
    img: [
        fruitsColor['peach.png'].default,
        fruitsLinear['peach.png'].default
    ]
}, {
    name: 'Pear',
    img: [
        fruitsColor['pear.png'].default,
        fruitsLinear['pear.png'].default
    ]
}, {
    name: 'Pineapple',
    img: [
        fruitsColor['pineapple.png'].default,
        fruitsLinear['pineapple.png'].default
    ]
}, {
    name: 'Plum',
    img: [
        fruitsColor['plum.png'].default,
        fruitsLinear['plum.png'].default
    ]
}, {
    name: 'Pomegranate',
    img: [
        fruitsColor['pomegranate.png'].default,
        fruitsLinear['pomegranate.png'].default
    ]
}, {
    name: 'Star Fruit',
    img: [
        fruitsColor['star-fruit.png'].default,
        fruitsLinear['star-fruit.png'].default
    ]
}, {
    name: 'Strawberry',
    img: [
        fruitsColor['strawberry.png'].default,
        fruitsLinear['strawberry.png'].default
    ]
}, {
    name: 'Watermelon',
    img: [
        fruitsColor['watermelon.png'].default,
        fruitsLinear['watermelon.png'].default
    ]
}]

export { fruits as default }