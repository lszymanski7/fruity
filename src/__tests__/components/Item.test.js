import React from 'react'
import { render, screen } from '@testing-library/react'
import Item from '../../components/Item'
import fruits from '../../data/fruits'

const props = {
    id: fruits[0].id,
    name: fruits[0].name,
    img: fruits[0].img,
    checked: fruits[0].checked,
    fruits,
    setFruits: jest.fn(),
    setFilteredFruits: jest.fn()
}

describe('Item', () => {
	it('should be rendered correctly.', () => {
		const component = render(<Item {...props} />)
		expect(component).toMatchSnapshot()
	})

	it('should have the correct image.', () => {
		render(<Item {...props} />)
		const img = screen.getByRole('img')
		expect(img).toHaveAttribute('src', props.img.color)
		expect(img).toHaveAttribute('alt', props.name)
		expect(img).toBeInTheDocument()
	})

    it('should have the correct title.', () => {
		render(<Item {...props} />)
		const title = props.name
		const h3 = screen.getByRole('heading', { level: 3 })
		expect(h3).toHaveTextContent(title)
	})
})
