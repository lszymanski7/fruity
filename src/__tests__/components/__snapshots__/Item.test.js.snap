// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Item should be rendered correctly. 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="item"
      >
        <img
          alt="Apple"
          class="item__image"
          id="bbb33b2d-e07d-4d13-a869-f3790e100477"
          src="apple-color.png"
        />
        <p
          class="item__title"
        >
          Apple
        </p>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="item"
    >
      <img
        alt="Apple"
        class="item__image"
        id="bbb33b2d-e07d-4d13-a869-f3790e100477"
        src="apple-color.png"
      />
      <p
        class="item__title"
      >
        Apple
      </p>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
